app/src/features/surveys/observations/sampling-sites/SamplingSiteList.tsx

```diff
@@ -9,173 +9,210 @@ import { grey } from '@mui/material/colors';
 import IconButton from '@mui/material/IconButton';
 import List from '@mui/material/List';
 import ListItem from '@mui/material/ListItem';
+import ListItemIcon from '@mui/material/ListItemIcon';
 import ListItemText from '@mui/material/ListItemText';
+import Menu, { MenuProps } from '@mui/material/Menu';
+import MenuItem from '@mui/material/MenuItem';
 import Toolbar from '@mui/material/Toolbar';
 import Typography from '@mui/material/Typography';
+import { CodesContext } from 'contexts/codesContext';
+import { SurveyContext } from 'contexts/surveyContext';
+import { useContext, useEffect, useState } from 'react';
 import { useHistory } from 'react-router';
+import { getCodesName } from 'utils/Utils';
 
 const SamplingSiteList = () => {
   const history = useHistory();
 
+  const surveyContext = useContext(SurveyContext);
+  const codesContext = useContext(CodesContext);
+
+  useEffect(() => {
+    codesContext.codesDataLoader.load();
+  }, [codesContext.codesDataLoader]);
+
+  surveyContext.sampleSiteDataLoader.load(surveyContext.projectId, surveyContext.surveyId);
+
+  const [anchorEl, setAnchorEl] = useState<MenuProps['anchorEl']>(null);
+  const [selectedSampleSiteId, setSelectedSampleSiteId] = useState<number | undefined>();
+
+  const handleMenuClick = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>, sample_site_id: number) => {
+    setAnchorEl(event.currentTarget);
+    setSelectedSampleSiteId(sample_site_id);
+  };
+
+  if (
+    !surveyContext.sampleSiteDataLoader.data ||
+    (surveyContext.sampleSiteDataLoader.isLoading && !codesContext.codesDataLoader.data) ||
+    codesContext.codesDataLoader.isLoading
+  ) {
+    // TODO add loading animation
+    return <></>;
+  }
+
   return (
-    <Box display="flex" flexDirection="column" height="100%">
-      <Toolbar
-        sx={{
-          flex: '0 0 auto',
-          borderBottom: '1px solid #ccc'
+    <>
+      <Menu
+        open={Boolean(anchorEl)}
+        onClose={() => setAnchorEl(null)}
+        anchorEl={anchorEl}
+        anchorOrigin={{
+          vertical: 'top',
+          horizontal: 'right'
+        }}
+        transformOrigin={{
+          vertical: 'top',
+          horizontal: 'right'
         }}>
-        <Typography
-          sx={{
-            flexGrow: '1'
+        <MenuItem
+          onClick={() => {
```

Comment:
This isn't a major issue, but exposing the link to the browser is a generally a good idea because a) a11y/semantics, b) users can preview where the link will take them before they go, and c) users get access to the appropriate context menu, so they can do things like _Open in new tab_, etc.

Reply:
updated